oak-blob-cloud-azure/pom.xml

```diff
@@ -50,8 +69,23 @@
                         <Embed-Dependency>
                             azure-storage,
                             azure-keyvault-core,
+                            azure-core,
+                            azure-identity,
+                            azure-json,
                             guava,
-                            jsr305
+                            jsr305,
```

Comment:
I'm very concerned about the amount of embeds here. Is this really needed? It will make the nundle bigger, and will make updates harder...

Reply:
At least JSR305 is never necessary at run time AFAIK. It is only annotations relevant at compile time.

Reply:
@reschke, we are using azure-identity library for migration to service principals from access key based authentication in azure. All these dependencies came transitively from azure-identity. I initially tried marking some of them as optional which helped in resolving the bundle to active state but when I configured the component to use service principals, it failed with NoClassDefFoundError exceptions meaning that they were indeed required at runtime and cannot be marked as optional.

Attaching the file of errors that I got and the corresponding dependency embedded.
[dependencies_list.pdf](https://github.com/apache/jackrabbit-oak/files/15319377/dependencies_list.pdf)

Reply:
@reschke @kwin do we have anything else to add regarding the embeds list? otherwise I'd suggest to go forward with the PR

cc @t-rana

Reply:
I just wanted to make sure we know what we do :-)